core/src/main/java/io/grpc/transport/WritableBufferAllocator.java

```diff
@@ -32,13 +32,14 @@
 package io.grpc.transport;
 
 /**
- * The preferred way of creating a {@link WritableBuffer}.
+ * An allocator of buffers provided by the transport implementation to {@link MessageFramer} so
+ * it can send chunks of data to the transport in a form that the transport can directly serialize.
  */
 public interface WritableBufferAllocator {
 
   /**
-   * Allocate a new {@link WritableBuffer} that can hold
-   * {@code capacity} bytes.
+   * Request a new {@link WritableBuffer} with the given {@code capacity}. The allocator is
+   * free to return a buffer with a greater or lesser capacity.
```

Comment:
That's not yet being done by this PR, correct?  Is the idea that `NettyWritableBufferAllocator` would be modified to be aware of the max frame size and therefore possibly provide a buffer smaller than what was requested?

Once flow control kicks in, however, all bets are off since whatever we pass to the transport will likely get chunked anyway.  From the perspective of the framer writing to the transport, bigger is always better ... which means we probably don't need to care too much about the current max frame size WRT the framer. Maybe we just always request a buffer that's something a bit larger (since they're pooled), like 1MB?